src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1423,6 +1433,11 @@ BoundExpression bindDefaultArgument(SyntaxNode syntax, ParameterSymbol parameter
                 }
 
                 return defaultValue;
+
+                static int getArgumentIndex(int parameterIndex, ImmutableArray<int> argsToParamsOpt)
+                    => argsToParamsOpt.IsDefault
```

Comment:
>argsToParamsOpt.IsDefault [](start = 23, length = 25)

Is this condition ever true?

Reply:
@AlekseyTs Yes. When the arguments order matches the parameters order. I don't know the reason, only figured that by testing.

Reply:
>> Is this condition ever true?

> Yes. When the arguments order matches the parameters order. I don't know the reason, only figured that by testing.

Just to confirm. We have omitted arguments and argsToParamsOpt.IsDefault can still be true?

---
In reply to: [600234291](https://github.com/dotnet/roslyn/pull/51952#discussion_r600234291) [](ancestors = 600234291)

Reply:
> We have omitted arguments and argsToParamsOpt.IsDefault can still be true? 

Not sure what you mean by "We have omitted arguments". But yes I confirmed it can be true.

For example, for the following test:

```csharp
            string source = @"
using System;
using System.Runtime.CompilerServices;

class Program
{
    public static void Main()
    {
        Log(123);
    }
    const string p = nameof(p);
    static void Log(int p, [CallerArgumentExpression(p)] string arg = null)
    {
        Console.WriteLine(arg);
    }
}
";
```

![image](https://user-images.githubusercontent.com/31348972/112349996-e3a18900-8cd1-11eb-890d-f3d2f5db2567.png)